File: solutions/javascript/pig-latin/1/pig-latin.js
//
// This is only a SKELETON file for the 'Pig Latin' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const translate = (text) => {
  const vowels = ['a', 'e', 'i', 'o', 'u'];

  const translateWord = (word) => {
    // Case 1: starts with vowel or "xr"/"yt"
    if (vowels.includes(word[0]) || word.startsWith("xr") || word.startsWith("yt")) {
      return word + "ay";
    }

    let consonantCluster = '';
    let i = 0;

    while (i < word.length && !vowels.includes(word[i])) {
      // Special "qu"
      if (word.startsWith("qu", i)) {
        consonantCluster += "qu";
        i += 2;
        continue;
      }

      // Special "y" (acts as vowel if not at position 0)
      if (word[i] === 'y' && i > 0) {
        break;
      }

      consonantCluster += word[i];
      i++;
    }

    return word.slice(i) + consonantCluster + "ay";
  };

  // Split sentence into words, translate each, rejoin
  return text.split(" ").map(translateWord).join(" ");
};

File: solutions/python/pig-latin/1/pig_latin.py
def translate(text):

    result = []
    vowels = ["a","e","i","o","u"]

    words = text.split()

    for word in words:

       
        if word[0] in vowels or word[0:2] == "xr" or word[0:2] == "yt":
            result.append(word + "ay")

        else:
            for i, ch in enumerate(word):

              
                if word[i:i+2] == "qu":
                    result.append(word[i+2:] + word[:i+2] + "ay")
                    break

              
                if ch in vowels or (ch == "y" and i != 0):
                    result.append(word[i:] + word[:i] + "ay")
                    break

    return " ".join(result)
